XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/UX/Scripts/Pointers/TeleportPointer.cs

```diff
@@ -212,24 +208,156 @@ public override void OnPostRaycast()
+                eventData.Handedness == Handedness &&
+                eventData.MixedRealityInputAction == MixedRealityToolkit.TeleportSystem.TeleportAction)
+            {
+                eventData.Use();
+                ProcessDigitalTeleportInput(true);
+            }
+        }
+
+        /// <inheritdoc />
+        public override void OnInputUp(InputEventData eventData)
+        {
+            if (eventData.SourceId == InputSourceParent.SourceId &&
+                eventData.Handedness == Handedness &&
+                eventData.MixedRealityInputAction == MixedRealityToolkit.TeleportSystem.TeleportAction)
+            {
+                eventData.Use();
+                ProcessDigitalTeleportInput(false);
+            }
+        }
+
+        /// <inheritdoc />
+        public override void OnInputChanged(InputEventData<float> eventData)
+        {
+            // Don't process input if we've got an active teleport request in progress.
+            if (eventData.used || IsTeleportRequestActive || !IsTeleportSystemEnabled)
+            {
+                return;
+            }
+
+            if (eventData.SourceId == InputSourceParent.SourceId &&
+                eventData.Handedness == Handedness &&
+                eventData.MixedRealityInputAction == MixedRealityToolkit.TeleportSystem.TeleportAction)
+            {
+                eventData.Use();
+                ProcessSingleAxisTeleportInput(eventData);
+            }
+        }
+
         /// <inheritdoc />
         public override void OnInputChanged(InputEventData<Vector2> eventData)
         {
             // Don't process input if we've got an active teleport request in progress.
-            if (IsTeleportRequestActive || !IsTeleportSystemEnabled) { return; }
+            if (eventData.used || IsTeleportRequestActive || !IsTeleportSystemEnabled)
+            {
+                return;
+            }
 
             if (eventData.SourceId == InputSourceParent.SourceId &&
                 eventData.Handedness == Handedness &&
-                eventData.MixedRealityInputAction == teleportAction)
+                eventData.MixedRealityInputAction == MixedRealityToolkit.TeleportSystem.TeleportAction)
+            {
+                eventData.Use();
+                ProcessDualAxisTeleportInput(eventData);
+            }
+        }
+
+        #endregion IMixedRealityInputHandler Implementation
+
+        #region IMixedRealityTeleportHandler Implementation
+
+        /// <inheritdoc />
+        public override void OnTeleportRequest(TeleportEventData eventData)
+        {
+            // Only turn off the pointer if we're not the one sending the request
+            if (eventData.Pointer.PointerId == PointerId)
+            {
+                IsTeleportRequestActive = false;
+            }
+            else
+            {
+                IsTeleportRequestActive = true;
+                BaseCursor?.SetVisibility(false);
```

Comment:
Let's do a traditional null check on the unity engine object types

Reply:
Agreed. I'll change that in another PR as this code was not introduced in this PR and I don't want to trigger another build on this one. Need this to release to continue testing.